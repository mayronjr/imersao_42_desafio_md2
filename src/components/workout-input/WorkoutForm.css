.WorkoutForm{
    display: flex;
    justify-content: space-around;
    margin-top: 5px;
    margin-bottom: 5px;
    user-select: none;
}