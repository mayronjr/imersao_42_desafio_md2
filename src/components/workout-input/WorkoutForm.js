import 'bootstrap/dist/css/bootstrap.min.css';
import './WorkoutForm.css';
import { Component } from 'react';

import getDateInInputFormat from './scripts.js';

class WorkoutForm extends Component {
    constructor(props){
        super(props)
        let aux = getDateInInputFormat()
        this.state = {
            valueInput1: 1,
            valueInput2: aux,
            min: aux,
            selectionList: ['Run', 'Swimming', 'Bike']
        }
    }

    handleValueInput1 = (e) =>{
        if(!isNaN(Number(e.target.value)) && e.target.value.length !== 0){
            this.setState({valueInput1: e.target.value})
        }
    }

    handleValueInput2 = (e)=>{
        if(e.target.value >= this.state.min){
            this.setState({
                valueInput2: e.target.value
            })
        }
    }

    handleSubmit=(e)=>{
        e.preventDefault()
        let date = e.target.date.value.split('-')
        let newWorkout = {
            tempo: Number(e.target.time.value),
            type: e.target.type.value,
            date: new Date(date[1] + '/' + date[2] + '/' + date[0])
        }
        if(this.props.onSubmit !== undefined){
            this.props.onSubmit(newWorkout)
        }
    }

    render(){
        let {valueInput1, valueInput2, min, selectionList} = this.state
        return(
            <form className="container WorkoutForm" onSubmit={this.handleSubmit.bind(this)} data-testid="form">
                <input name="time" type="number" min="1" value={valueInput1} onChange={this.handleValueInput1.bind(this)}/>
                <select name="type" id="type">
                    {selectionList.map((opt, _index)=>{
                        return(<option key={_index} value={opt}>{opt}</option>)
                    })}
                </select>
                <input name="date" type="date" min={min} value={valueInput2} onChange={this.handleValueInput2.bind(this)}/>
                <button type="submit" data-testid="form-submit">Add</button>
            </form>
        )
    }
}

export default WorkoutForm;