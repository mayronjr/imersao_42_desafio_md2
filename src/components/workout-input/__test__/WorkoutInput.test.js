import React from 'react'
import ReactDOM from 'react-dom'

import WorkoutForm from '../WorkoutForm.js'
import getDateInInputFormat from '../scripts.js';

import { render, cleanup, act } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"

import renderer from 'react-test-renderer'

const list = [
    { tempo: 1, type: 'Run', date: new Date(2011, 10, 24) },
    { tempo: 3, type: 'Swimming', date: new Date(2011, 8, 24) }
]

afterEach(cleanup)

it('renders without crashing', () => {
    const doc = document.createElement('div')
    ReactDOM.render(<WorkoutForm />, doc)
})

it('renders WorkoutInput Correctly', () => {
    const { getByTestId } = render(<WorkoutForm />)
    expect(getByTestId('form')).toHaveFormValues({
        time: 1,
        type: 'run',
        date: getDateInInputFormat()
    })
})

// it('submit Correctly', () => {
//     const callback = jest.fn()
//     let container = document.createElement('div')
//     act(()=>{
//         ReactDOM.render(<WorkoutForm onSubmit={callback}/>, container)
//     })
//     let submit = container.querySelector('form')
//     // expect(submit.innerHTML).toBe("Add")
//     act(()=>{
//         submit.dispatchEvent(new MouseEvent("click"), {bubbles: true})
//     })
//     expect(callback).toHaveBeenCalledTimes(1)
//     expect(callback).toHaveBeenCalledWith({
//         time: 1,
//         type: 'run',
//         date: getDateInInputFormat()
//     })
// })

it('matches snapshot', () => {
    const tree = renderer.create(<WorkoutForm/>).toJSON()
    expect(tree).toMatchSnapshot()
})