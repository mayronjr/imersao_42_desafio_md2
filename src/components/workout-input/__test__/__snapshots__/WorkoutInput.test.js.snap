// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className="container WorkoutForm"
  data-testid="form"
  onSubmit={[Function]}
>
  <input
    min="1"
    name="time"
    onChange={[Function]}
    type="number"
    value={1}
  />
  <select
    id="type"
    name="type"
  >
    <option
      value="Run"
    >
      Run
    </option>
    <option
      value="Swimming"
    >
      Swimming
    </option>
    <option
      value="Bike"
    >
      Bike
    </option>
  </select>
  <input
    min="2021-04-08"
    name="date"
    onChange={[Function]}
    type="date"
    value="2021-04-08"
  />
  <button
    data-testid="form-submit"
    type="submit"
  >
    Add
  </button>
</form>
`;
